File: 283.Move-Zeroes.py
# https://leetcode.com/problems/move-zeroes/description/
#
# algorithms
# Easy (52.73%)
# Total Accepted:    371.9K
# Total Submissions: 705.3K
# beats 100.0% of python submissions


class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        length = len(nums)
        if length <= 1:
            return

        idx = 0
        for n in nums:
            if n != 0:
                nums[idx] = n
                idx += 1

        for i in xrange(idx, length):
            nums[i] = 0

        return

File: 283.Move-Zeroes.java
// https://leetcode.com/problems/move-zeroes/
//
// algorithms
// Easy (54.08%)
// Total Accepted:    450,301
// Total Submissions: 832,623
// beats 100.0% of java submissions

class Solution {
    public void moveZeroes(int[] nums) {
        int idx = 0, i = 0, len = nums.length;
        
        for (; i < len; i++) {
            if (nums[i] != 0) {
                nums[idx] = nums[i];
                idx++;
            }
        }
        
        for (; idx < len; idx++) {
            nums[idx] = 0;
        }
        
    }
}